### PR TITLE
optimized complement and removed hooks

### PR DESCRIPTION
**Pull request by ebolyen**
A few more changes that have nothing to do with each other and this could be a congressional bill!

### CODE REVIEW

File: skbio/sequence/_nucleotide_sequence.py [hunk 1]
```diff
@@ -39,6 +41,18 @@ class NucleotideSequence(with_metaclass(ABCMeta, IUPACSequence)):
...
 
     """
+    __complement_lookup = None
+
+    @classproperty
+    def _complement_lookup(cls):
+        if cls.__complement_lookup is not None:
+            return cls.__complement_lookup
+
+        lookup = np.zeros(256, dtype=np.uint8)
```
Comment (jairideout) [inline skbio/sequence/_nucleotide_sequence.py]: use `IUPACSequence._number_of_extended_ascii_codes`

**Comment by jairideout** [comment]
done reviewing, one minor but important comment

File: skbio/sequence/_nucleotide_sequence.py (hunk 1 above)
Comment (ebolyen) [inline skbio/sequence/_nucleotide_sequence.py]: Thanks, forgot about this!

**Comment by jairideout** [comment]
:shipit:

**Pull request merged**